rfc/ipni-w3c.md

```diff
@@ -0,0 +1,346 @@
+# W3 IPNI Indexing RFC
+
+## Authors
+
+- [gammazero], [olizilla]
+
+## Goals
+
+A user can use a CID that identifies data stored on w3c to lookup the location where the data can be retrieved and content claims describing that data.
+
+The data's location can change and content claims about the data can change, at anytime during the data's storage lifetime, and queried for the data CIDs will be able to get the most locations and claims.
+
+This RFC describes how IPNI indexing can be used to lookup data locations and content claims for any CID within a CAR stored in w3c.
+
+## Abstract
+
+A user stores data as a CAR file. This CAR contains a set of multihashes for individual pieces of data within the CAR. 
+
+The user creates a an external CAR index [CARx-index](https://github.com/web3-storage/RFC/pull/9/files) that contains tuples of (CID, offset, length) of any items they intend to make individually retrievable from the car.
+
+With an [inclusion claim](https://hackmd.io/@gozala/content-claims?utm_source=preview-mode&utm_medium=rec#Inclusion-Claims), the user who created the CAR asserts that CAR contains a given set of multihashes, transitively, via the CARx-index.
+
+After storing the CAR and CARx-index, w3s creates a [location claim](https://hackmd.io/@gozala/content-claims?utm_source=preview-mode&utm_medium=rec#Location-Claims) for the CAR file and the CARx-index. Each location claim maps an item's CID to a URL where the item can be retrieved.
+
+w3s creates a Content Claims Bucket. This is a container, stored on w3s network, that holds all content claims associated with the stored data. The location claims and inclusion claim are written to the claims bucket. The claims bucket is then stored.
```

Comment:
This is a case but also something we’d like to move away from in favor of storing claims in user spaces instead. That way costs associated with storing and serving claims is legged by users as opposed to be unpredictable expense for a service

Reply:
I was thinking that a user's claims bucket would be stored in user space. Does that work?